androidApp/src/main/java/com/example/tasky/android/agenda/di/AgendaModule.kt

```diff
@@ -0,0 +1,17 @@
+package com.example.tasky.android.agenda.di
+
+import com.example.tasky.android.agenda.viewmodel.AgendaViewModel
+import com.example.tasky.repository.AgendaRepository
+import com.example.tasky.repository.IAgendaRepository
+import org.koin.core.module.dsl.bind
+import org.koin.core.module.dsl.singleOf
+import org.koin.core.module.dsl.viewModelOf
+import org.koin.dsl.module
+
+val agendaModule =
+    module {
+        singleOf<AgendaRepository>(::AgendaRepository) {
+            bind<IAgendaRepository>()
+        }
```

Comment:
The pattern is `singleOf(::MyImplementation).bind<MyInterface>()`

In this case, you can change it to

```
singleOf(::AgendaRepository).bind<IAgendaRepository>()
```